openvdb/tools/FastSweeping.h

```diff
@@ -0,0 +1,1432 @@
+///          active tiles in the input grid will be converted to active voxels
+///          in the output grids!
+///
+/// @throw  RuntimeError if the iso-surface does not intersected any active
+///         voxels or if it intersects any active tiles in @a fogGrid.
+template<typename GridT, typename OpT>
+std::array<typename GridT::Ptr, 2>
+fogToSdfAndExt(const GridT &fogGrid,
+               const OpT &op,
+               typename GridT::ValueType isoValue,
+               int nIter = 1);
+
+/// @brief Computes the signed distance field and the extension of a scalar field,
+///        defined by the specified functor, off an iso-surface from an input SDF volume.
+///
+/// @return An array of two shared pointers to respectively the SDF and extension field
+///
+/// @param sdfGrid  Scalar (floating-point) volume from which an
+///                 iso-surface can be defined.
+///
+/// @param op       Functor with signature [](const Vec3R &xyz)->float that
+///                 defines the Dirichlet boundary condition, on the iso-surface,
+///                 of the field to be extended.
+///
+/// @param isoValue A value which defines a smooth iso-surface that
+///                 intersects active voxels in @a sdfGrid.
+///
+/// @param nIter    Number of iterations of the fast sweeping algorithm.
+///                 each performing 2^3 = 8 sweeps.
+///
+/// @note Strictly speaking a fog volume is normalized to the range [0,1] but this
+///       method accepts a scalar volume with an arbitary range, as long as the it
+///       includes the @a isoValue.
+///
+/// @details Topology of output grids are identical to that of the input grid, except
+///          active tiles in the input grid will be converted to active voxels
+///          in the output grids!
+///
+/// @throw  RuntimeError if the iso-surface does not intersected any active
+///         voxels or if it intersects any active tiles in @a fogGrid.
+template<typename GridT, typename OpT>
+std::array<typename GridT::Ptr, 2>
+sdfToSdfAndExt(const GridT &sdfGrid,
+               const OpT &op,
+               typename GridT::ValueType isoValue = 0,
+               int nIter = 1);
+
+/// @brief Dilates an existing signed distance filed by a specified number of voxels
+///
+/// @return A shared pointer to the dilated signed distance field.
+///
+/// @param sdfGrid  Input signed distance field to to be dilated.
+///
+/// @param dilation Numer of voxels that the input SDF will be dilated.
+///
+/// @param nn       Stencil-pattern used for dilation
+///
+/// @param nIter    Number of iterations of the fast sweeping algorithm.
+///                 each performing 2^3 = 8 sweeps.
+///
+/// @details Topology will change as a result of the dilation
+template<typename GridT>
+typename GridT::Ptr
+dilateSdf(const GridT &sdfGrid,
+          int dilation,
+          NearestNeighbors nn = NN_FACE,
+          int nIter = 1);
+
+/// @brief Expand an existing signed distance fild into a mask
```

Comment:
Maybe:

Fills mask by extending an existing signed distance field into it.

Reply:
I'll adopt your version